### DISCUSSION TITLE
Support for aggregation queries?

### DISCUSSION THREAD

**Discussion by Cry0nicS** [vuejs/vuefire#1546] [Questions and Help] [answered]
I was trying to build some sort of pagination for a collection, when I realized there's no way to easily get a count of all the records in a collection.

It looks like Firebase introduced recently these "[aggregation queries](https://firebase.google.com/docs/firestore/query-data/aggregation-queries)" that allows for exactly such a scenario.

Is there any support planned for these? Or is there a way to already make sure of this?

**Comment by posva** [comment, marked as answer]
You don't need anything specific from vuefire to use aggregated queries. Since they return a promise, you can already use them directly. e.g. in Nuxt, you can use `await` or https://nuxt.com/docs/api/composables/use-async-data. Or you can use [data loaders](https://uvr.esm.is/rfcs/data-loaders/). It's really up to you!

**Reply by Cry0nicS** [reply]
> You don't need anything specific from vuefire to use aggregated queries. Since they return a promise, you can already use them directly. e.g. in Nuxt, you can use `await` or https://nuxt.com/docs/api/composables/use-async-data. Or you can use [data loaders](https://uvr.esm.is/rfcs/data-loaders/). It's really up to you!

@posva I see what you mean. However, the main issue with `await` or `useAsyncData` is that I have to fetch ALL items. 

Say I have like 1000 items, and I want to paginate them 20 at a time.
I can use the `limit` for the `query` and only fetch 20. I would still, however, need to know how many items I have in total so I can display the 50 pages necessary...
Fetching all 1000 records only to display 20 is not great

The `count()` aggregation would allow me to use `getCountFromServer` (that was just an idea, still not sure what's the best way to handle this)
